airbyte-webapp/src/views/Connector/ConnectorCard/useConnectorCardService.tsx

```diff
@@ -0,0 +1,200 @@
+import { JSONSchema7 } from "json-schema";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { AnySchema } from "yup";
+
+import { Connector, ConnectorSpecification } from "core/domain/connector";
+import { CheckConnectionRead, SynchronousJobRead } from "core/request/AirbyteClient";
+import { LogsRequestError } from "core/request/LogsRequestError";
+import { useAdvancedModeSetting } from "hooks/services/useAdvancedModeSetting";
+import { ServiceFormValues } from "views/Connector/ServiceForm";
+
+import {
+  FormBaseItem,
+  FormBlock,
+  FormComponentOverrideProps,
+  WidgetConfig,
+  WidgetConfigMap,
+} from "../../../core/form/types";
+import { useFormChangeTrackerService, useUniqueFormId } from "../../../hooks/services/FormChangeTracker";
+import { ConnectorNameControl } from "../ServiceForm/components/Controls/ConnectorNameControl";
+import {
+  useBuildForm,
+  useBuildInitialSchema,
+  useBuildUiWidgetsContext,
+  useConstructValidationSchema,
+} from "../ServiceForm/useBuildForm";
+import { ConnectorCardProps } from "./interfaces";
+import { useAnalyticsTrackFunctions } from "./useAnalyticsTrackFunctions";
+import { useTestConnector } from "./useTestConnector";
+
+interface UseConnectorCardServiceHookResult {
+  advancedMode: boolean;
+  isFormSubmitting: boolean;
+  isTestConnectionInProgress: boolean;
+  job?: SynchronousJobRead | null;
+  onHandleSubmit: (values: ServiceFormValues) => Promise<void>;
+  onStopTesting: () => void;
+  saved: boolean;
+  testConnector: (v?: ServiceFormValues) => Promise<CheckConnectionRead>;
+  error: Error | null;
+  selectedConnectorDefinitionSpecificationId?: string;
+  uniqueFormId?: string;
+  jsonSchema: JSONSchema7;
+  initialValues: ServiceFormValues;
+  formFields: FormBlock;
+  validationSchema: AnySchema;
+  uiWidgetsInfo: WidgetConfigMap;
+  setUiWidgetsInfo: (widgetId: string, updatedValues: WidgetConfig) => void;
+  resetUiWidgetsInfo: () => void;
+  getValues: <T = unknown>(values: ServiceFormValues<T>) => ServiceFormValues<T>;
+  onFormSubmit: (values: ServiceFormValues) => Promise<void>;
+}
```

Comment:
While I do think this refactor has made the ServiceForm and ConnectorCard components more readable, it feels like all of their complexity has just been shoved into this service instead.

Could you try to simplify this or break it into smaller services? It feels like there is a ton going on here which makes this service pretty hard to understand on its own without going and seeing how each result value is used.

For example, this returns an `onHandleSubmit` and an `onFormSubmit`, both of which have the same input and return types -- why are both necessary?

It feels like this service could be split up into more readable pieces. Maybe all of the `uiWidget` logic can be contained in a single place, and/or maybe more comments/documentation could be added here to help make it easier to understand.

Reply:
yeah, that makes sense. I am currently working on this, but still a bit hard, assuming I am not familiar with logic inside this component at all

Reply:
@lmossman you can check now, I added two new smaller hooks that split some logic from `useConnectorCardService`, and are used inside the original hook